### PR TITLE
neocities-cli: init at 0.0.18

### PR DESCRIPTION
**Pull request by Dawoox**
## Description of changes

Inits [neocities-cli](https://neocities.org/cli), a cli to interact with your neocities website from the terminal.

20240222_01h09m28s_grim

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ne/neocities-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  dontUnpack = true;
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ${ttBundlerApp}/bin/${name} $out/bin
+  '';
+
+  meta = with lib; {
```
Comment (kirillrdy) [inline pkgs/by-name/ne/neocities-cli/package.nix:29]: consider adding mainProgram

File: pkgs/by-name/ne/neocities-cli/Gemfile.lock [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      pastel (~> 0.7.2)
+      strings (~> 0.1.0)
+      tty-screen (~> 0.6.4)
+    unicode-display_width (1.8.0)
+    unicode_utils (1.4.0)
+    wisper (1.6.1)
+
+PLATFORMS
+  ruby
+  x86_64-linux
```
Comment (kirillrdy) [inline pkgs/by-name/ne/neocities-cli/Gemfile.lock:42]: bundix doesn't deal well with platform gems,  maybe it's ok in this case. Is there an upstream Gemfile.lock ?

File: pkgs/by-name/ne/neocities-cli/package.nix (hunk 1 above)
Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/package.nix:29]: done

File: pkgs/by-name/ne/neocities-cli/Gemfile.lock (hunk 2 above)
Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/Gemfile.lock:42]: As far as I know, no it doesn't seem

File: pkgs/by-name/ne/neocities-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    description = "neocities";
+    homepage = "https://github.com/neocities/neocities-ruby";
+    mainProgram = "neocities";
```
[the remark this answers is not in the document]
Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/package.nix]: In the last commit i've set it as `mainProgram = name`.

File: pkgs/by-name/ne/neocities-cli/package.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    pname = "neocities";
+    gemdir = ./.;
+    exes = [ pname ];
+
+    passthru.updateScript = bundlerUpdateScript "neocities";
+  };
+in
+
+stdenv.mkDerivation rec {
+  name = "neocities";
```
Comment (kirillrdy) [inline pkgs/by-name/ne/neocities-cli/package.nix:19]: this should be `pname`
and `version` should be set

File: pkgs/by-name/ne/neocities-cli/package.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ${ttBundlerApp}/bin/${name} $out/bin
+  '';
+
+  meta = with lib; {
+    description = "The Neocities Gem - A CLI and library for using the Neocities web site API.";
+    homepage = "https://github.com/neocities/neocities-ruby";
+    mainProgram = name;
```
Comment (kirillrdy) [inline pkgs/by-name/ne/neocities-cli/package.nix:32]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

> Always prefer using a hardcoded string (don't use pname, for example).

File: pkgs/by-name/ne/neocities-cli/package.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{ stdenv
+, lib
+, bundlerApp
+, bundlerUpdateScript
+, installShellFiles
+}:
+
+let
+  ttBundlerApp = bundlerApp rec {
```
Comment (kirillrdy) [inline pkgs/by-name/ne/neocities-cli/package.nix:9]: not sure why not just use bundlerApp as final derivation ?

File: pkgs/by-name/ne/neocities-cli/package.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    pname = "neocities";
+    gemdir = ./.;
+    exes = [ pname ];
```
Comment (kirillrdy) [inline pkgs/by-name/ne/neocities-cli/package.nix:12]: more on pname

https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/ne/neocities-cli/package.nix (hunk 5 above)
Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/package.nix:32]: I changed this based on @Sigmanificient message, shouldn't have. I'm going to reverse it

Comment (kirillrdy) [reply pkgs/by-name/ne/neocities-cli/package.nix:32]: no worries ! nixpkgs is a huge project, lots of things to learn

File: pkgs/by-name/ne/neocities-cli/package.nix (hunk 7 above)
Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/package.nix:12]: so should I just hardcode `exes` attribute ?

Comment (kirillrdy) [reply pkgs/by-name/ne/neocities-cli/package.nix:12]: yes

have a look at these

```
[kirillvr@tsutenkaku:~/nixpkgs/pkgs/by-name]$ rg bundlerApp
pd/pdk/package.nix
ma/maid/package.nix
ba/bashly/package.nix
pr/pru/package.nix
yo/youplot/package.nix
```

File: pkgs/by-name/ne/neocities-cli/package.nix (hunk 6 above)
Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/package.nix:9]: Yes, I just realize I can cut in half the size of the file quite easily.

File: pkgs/by-name/ne/neocities-cli/package.nix (hunk 7 above)
Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/package.nix]: I've done that in the last commit, let me know if it's right

File: pkgs/by-name/ne/neocities-cli/package.nix (hunk 5 above)
Comment (Sigmanificient) [reply pkgs/by-name/ne/neocities-cli/package.nix]: :<

**Comment by kirillrdy** [comment]
please squash your commits into two

`maintainers: add dawoox`
`neocities-cli: init at 0.0.18` and PR title should also be `neocities-cli: init at 0.0.18`

**Comment by kirillrdy** [comment]
the order of commits is important
otherwise at `neocities-cli: init at 0.0.18`

```
neocities-cli.meta.maintainers
[ error: undefined variable 'dawoox'

```

**Comment by Dawoox** [comment]
That should be fixed now

File: pkgs/by-name/ne/neocities-cli/package.nix [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+}:
+
+bundlerApp {
+  pname = "neocities";
+  gemdir = ./.;
+  exes = [ "neocities" ];
+
+  passthru.updateScript = bundlerUpdateScript "neocities";
+
+  meta = with lib; {
```
[the remark this answers is not in the document]
Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/package.nix:13]: No problem, to be honest if my code was better written you would take way less time to verify my PR each time

File: pkgs/by-name/ne/neocities-cli/package.nix [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    description = "The Neocities Gem - A CLI and library for using the Neocities web site API.";
+    homepage = "https://github.com/neocities/neocities-ruby";
+    license = licenses.unfree;
```
Comment (kirillrdy) [inline pkgs/by-name/ne/neocities-cli/package.nix:16]: I just want to double check, if you are ok with sticking with this, there are implications of using unfree license, eg ofborg and hydra will not build such derivations.

Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/package.nix:16]: I'm ok with that, on the main github repo of the neocities cli there is no license so I guess I don't have the choice anyway.

Comment (kirillrdy) [reply pkgs/by-name/ne/neocities-cli/package.nix:16]: how about we merge this, but you follow up with upstream about setting some license ? then we can update it once upstream resolves it

Comment (Dawoox) [reply pkgs/by-name/ne/neocities-cli/package.nix:16]: After digging a bit further in the source code of the project I found that in the Gemspec `spec.license = "MIT"`.
Should we consider that as enough to mark it with the MIT license, or do you still want me to contact upstream to get a definitive answer ?

Comment (kirillrdy) [reply pkgs/by-name/ne/neocities-cli/package.nix:16]: thats good enough for me, as thats what https://rubygems.org/ would use

**Comment by kirillrdy** [comment]
@Dawoox thank you very much

we can resolve other issues in subsequent PR's

**Pull request merged**